### PR TITLE
xDai Implementation

### PR DESCRIPTION
**Pull request by jacobcreech**
## Description

Added xDai Implementation

## Testing instructions

Run `./bootstrap.sh` or `make -C build tests`

## Types of changes

Added xDai Support. Related to issue #80 

## Checklist

- [X] Prefix PR title with `[WIP]` if necessary.
- [X] Add tests to cover changes as needed.
- [X] Update documentation as needed.

### CODE REVIEW

File: include/TrustWalletCore/TWEthereumChainID.h [hunk 1]
```diff
@@ -20,6 +20,7 @@ enum TWEthereumChainID {
     TWEthereumChainIDVeChain = 74,
     TWEthereumChainIDThunderToken = 18,
     TWEthereumChainIDTomoChain = 88,
+    TWEthereumChainIDDai = 100,
```
Comment (vikmeup) [inline include/TrustWalletCore/TWEthereumChainID.h]: where does this number come from? xdai is 800 ID on slip 44: https://github.com/satoshilabs/slips/blob/master/slip-0044.md

File: src/interface/TWCoinTypeConfiguration.cpp [hunk 2]
```diff
@@ -31,6 +31,7 @@ TWString *_Nullable TWCoinTypeConfigurationGetSymbol(enum TWCoinType type) {
     case TWCoinTypeTron: string = "TRX"; break;
     case TWCoinTypeVeChain: string = "VET"; break;
     case TWCoinTypeWanChain: string = "WAN"; break;
+    case TWCoinTypeDai: string = "DAI"; break;
```
Comment (vikmeup) [inline src/interface/TWCoinTypeConfiguration.cpp]: should be `xDAI` ?

File: src/interface/TWCoinTypeConfiguration.cpp [hunk 3]
```diff
@@ -173,6 +177,7 @@ TWString *_Nonnull TWCoinTypeConfigurationGetID(enum TWCoinType type) {
     case TWCoinTypeTron: string = "tron"; break;
     case TWCoinTypeVeChain: string = "vechain"; break;
     case TWCoinTypeWanChain: string = "wanchain"; break;
+    case TWCoinTypeDai: string = "dai"; break;
```
Comment (vikmeup) [inline src/interface/TWCoinTypeConfiguration.cpp]: `xdai`

File: src/interface/TWCoinTypeConfiguration.cpp [hunk 4]
```diff
@@ -203,6 +208,7 @@ TWString *_Nonnull TWCoinTypeConfigurationGetName(enum TWCoinType type) {
     case TWCoinTypeTron: string = "Tron"; break;
     case TWCoinTypeVeChain: string = "VeChain"; break;
     case TWCoinTypeWanChain: string = "Wanchain"; break;
+    case TWCoinTypeDai: string = "Dai"; break;
```
Comment (vikmeup) [inline src/interface/TWCoinTypeConfiguration.cpp]: `xDAI`

File: include/TrustWalletCore/TWEthereumChainID.h (hunk 1 above)
Comment (jacobcreech) [reply include/TrustWalletCore/TWEthereumChainID.h]: This number comes from the original issue, where the information was given that xdai was chainId 100. You can find the same information here: https://forum.poa.network/t/xdai-chain-resources/1769

Comment (hewigovens) [reply include/TrustWalletCore/TWEthereumChainID.h]: @vikmeup it's chain id not coin type

File: include/TrustWalletCore/TWCoinType.h [hunk 5]
```diff
@@ -37,6 +37,7 @@ enum TWCoinType {
     TWCoinTypeTron = 195,
     TWCoinTypeVeChain = 818,
     TWCoinTypeWanChain = 5718350,
+    TWCoinTypeDai = 100,
```
Comment (hewigovens) [inline include/TrustWalletCore/TWCoinType.h]: should be `700` https://github.com/satoshilabs/slips/blob/master/slip-0044.md

**Comment by jacobcreech** [comment]
> CI is failed, could you please run `tools/ios-test` and `tools/android-test` and fix it?

Will do

**Comment by jacobcreech** [comment]
@igorbarinov I have resolved merge conflicts brought in by the stellar integration getting pulled. Let me know if any other conflicts come up.

**Pull request merged**